npm/test-vm.js

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env node
+// ---------------------------------------------------------------------------------------------------------------------
+// This script is intended to execute all unit tests in the Node VM: https://nodejs.org/api/vm.html.
+// ---------------------------------------------------------------------------------------------------------------------
+/* eslint-env node, es6 */
+
+require('shelljs/global');
+require('colors');
+
+// set directories and files for test and coverage report
+var vm = require('vm'),
+    path = require('path'),
+
+    Mocha = require('mocha'),
+    recursive = require('recursive-readdir'),
+
+    SPEC_SOURCE_DIR = path.join(__dirname, '..', 'test', 'unit');
+
+module.exports = function (exit) {
+    // banner line
+    console.log('Running integration tests using mocha in the node vm...'.yellow.bold);
+
+    // add all spec files to mocha
+    recursive(SPEC_SOURCE_DIR, function (err, files) {
+        if (err) { console.error(err); return exit(1); }
+
+        var mocha = new Mocha({timeout: 1000 * 60});
+
+        // specially load bootstrap file
+        mocha.addFile(path.join(SPEC_SOURCE_DIR, '_bootstrap.js'));
+
+        files.filter(function (file) { // extract all test files
+            return (file.substr(-8) === '.test.js');
+        }).forEach(mocha.addFile.bind(mocha));
+
+        // run loaded tests in the Node VM
+        vm.runInContext(`mocha.run(function (err) {
```

Comment:
This `require`s mocha from the outside the VM. All the sandbox require's are processed outside the VM, so this doesn't truly test the functionality inside the VM. 

Ideally, what you want is:

1. Create a VM context
2. Run the bundle in the context
3. Then load and run tests in the same VM context